### PR TITLE
Allow workflow plugins to specify their input and output ports (CMEM-4791)

### PR DESCRIPTION
**Pull request by robertisele**

### CODE REVIEW

File: cmem_plugin_base/dataintegration/plugins.py [hunk 1]
```diff
@@ -51,6 +52,13 @@ class PluginBase:
 class WorkflowPlugin(PluginBase):
     """Base class of all workflow operator plugins."""
 
+    input_ports: InputPorts
```
Comment (seebi) [inline cmem_plugin_base/dataintegration/plugins.py:55]: @robertisele Input Ports are mandatory?

File: cmem_plugin_base/dataintegration/ports.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+"""Workflow operator input and output ports."""
+
+from typing import Sequence
+from cmem_plugin_base.dataintegration.entity import EntitySchema
+
+
+class Port:
```
Comment (seebi) [inline cmem_plugin_base/dataintegration/ports.py:7]: @robertisele How can we specify a Port which does not accept data ... is used just for execution order?

Comment (seebi) [reply cmem_plugin_base/dataintegration/ports.py:7]: ok, remember - we will do a separate dependency port ...

File: cmem_plugin_base/dataintegration/plugins.py (hunk 1 above)
Comment (robertisele) [reply cmem_plugin_base/dataintegration/plugins.py:57]: Yes, declaring input and output ports are mandatory. But if the user still doesn't declare those it defaults to this:
```
    op.inputPorts shouldBe FlexibleNumberOfInputs()
    op.outputPort shouldBe Some(UnknownSchemaPort)
```

**Pull request merged**